.ci/packaging.groovy

```diff
@@ -154,6 +128,61 @@ pipeline {
   }
 }
 
+def releaseManager(def args = [:]) {
+  def bucketLocation = getBucketLocation(args.type)
+  googleStorageDownload(bucketUri: "${bucketLocation}/*",
+                        credentialsId: "${JOB_GCS_CREDENTIALS}",
+                        localDirectory: "${BASE_DIR}/build/distributions",
+                        pathPrefix: env.PATH_PREFIX)
+  dir("${BASE_DIR}") {
+    dockerLogin(secret: env.DOCKER_SECRET, registry: env.DOCKER_REGISTRY)
+    script {
+      getVaultSecret.readSecretWrapper {
+        sh(label: 'release-manager.sh', script: ".ci/scripts/release-manager.sh ${args.type}")
+      }
+    }
+  }
+}
+
+def package(def args = [:]) {
+  def type = args.type
+  def makeGoal = 'release-manager-snapshot'
+  if (type.equals('staging')) {
+    makeGoal = 'release-manager-release'
+  }
+  deleteDir()
+  unstash 'source'
+  dir("${BASE_DIR}"){
+    withMageEnv() {
+      sh(label: 'make release-manager', script: "make ${makeGoal}")
+    }
+  }
+}
+
+def publishArtifacts(def args = [:]) {
+  def bucketLocation = getBucketLocation(args.type)
+  // Copy those files to another location with the sha commit to test them afterward.
+  googleStorageUpload(bucket: "${bucketLocation}",
+    credentialsId: "${JOB_GCS_CREDENTIALS}",
+    pathPrefix: "${BASE_DIR}/build/distributions/",
+    pattern: "${BASE_DIR}/build/distributions/**/*",
+    sharedPublicly: true,
+    showInline: true)
+  // Copy the dependencies files if no ARM
+  whenFalse(isArm()) {
+    googleStorageUpload(bucket: "${bucketLocation}",
+      credentialsId: "${JOB_GCS_CREDENTIALS}",
+      pathPrefix: "${BASE_DIR}/build/",
+      pattern: "${BASE_DIR}/build/dependencies.csv",
+      sharedPublicly: true,
+      showInline: true)
+  }
+}
+
+def getBucketLocation(type) {
+  return "gs://${JOB_GCS_BUCKET}/${URI_SUFFIX}/${type}"
```

Comment:
Artifacts are not stored in either the `staging` or `snapshot` subfolders, this is quite handy in order to run the same workflow for different type of releases